libgnucash/engine/qof-string-cache.cpp

```diff
@@ -43,64 +45,42 @@ static QofLogModule log_module = QOF_MOD_UTIL;
 /* =================================================================== */
 /* The QOF string cache                                                */
 /*                                                                     */
-/* The cache is a GHashTable where a copy of the string is the key,    */
-/* and a ref count is the value                                        */
+/* The cache is an unordered_map where a copy of the string is the     */
+/* key, and a ref count is the value                                   */
 /* =================================================================== */
 
-static GHashTable* qof_string_cache = NULL;
-
-static GHashTable*
-qof_get_string_cache(void)
-{
-    if (!qof_string_cache)
-    {
-        qof_string_cache = g_hash_table_new_full(
-                               g_str_hash,               /* hash_func          */
-                               g_str_equal,              /* key_equal_func     */
-                               g_free,                   /* key_destroy_func   */
-                               g_free);                  /* value_destroy_func */
-    }
-    return qof_string_cache;
-}
+static std::unordered_map<std::string, guint> qof_string_cache;
 
 void
 qof_string_cache_init(void)
 {
-    (void)qof_get_string_cache();
+    qof_string_cache.clear();
 }
 
 void
 qof_string_cache_destroy (void)
 {
-    if (qof_string_cache)
-    {
-        g_hash_table_destroy(qof_string_cache);
-    }
-    qof_string_cache = NULL;
+    qof_string_cache.clear();
 }
```

Comment:
Same here. A container cleans up after itself when it goes out of scope, so I don't think having an explicit destroy function that's called exactly once when the application quits is not very useful.